Sources/OpenCombine/Publishers/Publishers.MapError.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  File.swift
+//  
+//
+//  Created by Joseph Spadafora on 7/4/19.
+//
+
+import Foundation
+
+extension Publishers {
+
+    /// A publisher that converts any failure from the
+    /// upstream publisher into a new error.
+    public struct MapError<Upstream: Publisher, Failure: Error>: Publisher {
+
+        /// The kind of values published by this publisher.
+        public typealias Output = Upstream.Output
+
+        /// The publisher from which this publisher receives elements.
+        public let upstream: Upstream
+
+        /// The closure that converts the upstream failure into a new error.
+        public let transform: (Upstream.Failure) -> Failure
+
+        public init(upstream: Upstream, _ map: @escaping (Upstream.Failure) -> Failure) {
+            self.upstream = upstream
+            self.transform = map
+        }
+
+        /// This function is called to attach the specified `Subscriber`
+        /// to this `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<SubscriberType: Subscriber>(subscriber: SubscriberType)
+            where Failure == SubscriberType.Failure,
+                  Upstream.Output == SubscriberType.Input
+        {
+            let mapErrorSubscriber = _MapError<Upstream, SubscriberType>(
+                downstream: subscriber,
+                transform: transform
+            )
+            upstream.receive(subscriber: mapErrorSubscriber)
+        }
+    }
+}
+
+extension Publisher {
+
+    /// Converts any failure from the upstream publisher into a new error.
+    ///
+    /// Until the upstream publisher finishes normally or fails with an error
+    /// , the returned publisher republishes all the elements it receives.
+    ///
+    /// - Parameter transform: A closure that takes the upstream failure as a
+    /// parameter and returns a new error for the publisher to terminate with.
+    /// - Returns: A publisher that replaces any upstream failure with a
+    /// new error produced by the `transform` closure.
+    public func mapError<ErrorType: Error>(
```

Comment:
I suggest to rename `ErrorType` to `NewFailure`, I feel that it adds some clarity that it's not just some arbitrary error, but a `Failure` type that will be used by the downstream